File: python/darts/darts.py
import math

def score(x, y):
    distance = distance_between_points(0, 0, x, y)

    if distance <= 1:
        return 10

    if distance <= 5:
        return 5

    if distance <= 10:
        return 1

    return 0

def distance_between_points(x_1, y_1, x_2, y_2):
    return math.sqrt(((x_2 - x_1) ** 2) + ((y_2 - y_1) ** 2))

File: c/darts/darts.c
#include "darts.h"


uint8_t score(coordinate_t landing_position) {

    float distance = distance_between_points(0, 0, landing_position.x, landing_position.y);
    return distance <= 1 ? 10
        : distance <= 5 ? 5
        : distance <= 10 ? 1
        : 0;
}

float distance_between_points(float x1, float y1, float x2, float y2) {

    return sqrt(pow((x2 - x1), 2) + pow((y2 - y1), 2));
}
